src/test/app/ValidatorKeys_test.cpp

```diff
@@ -30,149 +30,182 @@ namespace test {
+                "xQzlnVkZLaWxHZncxL3ZDeEhYWExwbGMyR25NaEFrRTFhZ3FYeEJ3RHdE\t  "
+                "\t\n",
+                "YklENk9NU1l1TTBGREFscEFnTms4U0tGbjdNTzJmZGtjd1JRSWhBT25ndTl\n",
+                "zQUtxWFlvdUorbDJWMFcrc0FPa1ZCK1pSUzZQU2hsSkFmVXNYZkFpQnNWSk\n",
+                "dlc2FhZE9KYy9hQVpva1MxdnltR21WcmxIUEtXWDNZeXd1NmluOEhBU1FLU\n",
+                "HVnQkQ2N2tNYVJGR3ZtcEFUSGxHS0pkdkRGbFdQWXk1QXFEZWRGdjVUSmEy\n",
+                "dzBpMjFlcTNNWXl3TFZKWm5GT3I3QzBrdzJBaVR6U0NqSXpkaXRROD0ifQ==\n"
+            };
+
+            const std::string tokenManifest =
+                "JAAAAAFxIe1FtwmimvGtH2iCcMJqC9gVFKilGfw1/vCxHXXLplc2GnMhAkE1ag"
+                "qXxBwDwDbID6OMSYuM0FDAlpAgNk8SKFn7MO2fdkcwRQIhAOngu9sAKqXYouJ+"
+                "l2V0W+sAOkVB+ZRS6PShlJAfUsXfAiBsVJGesaadOJc/aAZokS1vymGmVrlHPK"
+                "WX3Yywu6in8HASQKPugBD67kMaRFGvmpATHlGKJdvDFlWPYy5AqDedFv5TJa2w"
+                "0i21eq3MYywLVJZnFOr7C0kw2AiTzSCjIzditQ8=";
+
+            testValidatorToken(KeyType::secp256k1, tokenManifest, tokenBlob);
+
+            // Cannot specify both seed and token
+            testMakeValidatorKeys({
+                {SECTION_VALIDATION_SEED, {seed}},
+                {SECTION_VALIDATOR_TOKEN, {tokenBlob}}},
+                false /* expectValid */);
         }
 
         {
-            // Token manifest and private key must match
-            Config c;
-            c.section(SECTION_VALIDATOR_TOKEN).append(invalidTokenBlob);
-            ValidatorKeys k{c, j};
-
-            BEAST_EXPECT(k.configInvalid());
-            BEAST_EXPECT(k.publicKey.size() == 0);
-            BEAST_EXPECT(k.manifest.empty());
+            // validator token with ed25519 key
+            const std::vector<std::string> tokenBlob = {
+                "eyJtYW5pZmVzdCI6IkpBQUFBQUp4SWUzbWVnMnh1aTMxWWhyTDMvOHpCOHE\n",
+                "3SWduaXJ2d2xQTTlqeDBoZy8zODV1bk1oN2ZYUUdaU0l6SjlLMGJFaFNlMj\n",
+                "B4RjRrdTcvVFlFVXJTbkFlQnRpZitjVUhka0FXZFlJQU5xMTN4WUZ1blEvV\n",
+                "ElsSFhPeVRSZTgxc2gvbWVQV3pyU2djeFFMMlhIS0VuKzFIdkJWcnRXZXZ1\n",
+                "ay9hUUNqL0pMMnF0Z05KdFlwVkxIUndHY0JKQUM1aWpxMkJVREdidUxvRGZ\n",
+                "xZmRCUVpsMERQdTgzcElzNDlsUXNHKzI5eXZsZmxBcDhCVjd3UE9HK0hYMk\n",
+                "F6d2gzd3FzbXRodURlTENNM25WS1hOZkN3PT0iLCJ2YWxpZGF0aW9uX3NlY\n",
+                "3JldF9rZXkiOiI5RUQ0NUY4NjYyNDFDQzE4QTI3NDdCNTQzODdDMDYyNTkw\n",
+                "Nzk3MkY0RTcxOTAyMzFGQUE5Mzc0NTdGQTlEQUY2In0=\n"};
+
+            const std::string tokenManifest =
+                "JAAAAAJxIe3meg2xui31YhrL3/8zB8q7IgnirvwlPM9jx0hg/385unMh7fX"
+                "QGZSIzJ9K0bEhSe20xF4ku7/TYEUrSnAeBtif+cUHdkAWdYIANq13xYFunQ"
+                "/TIlHXOyTRe81sh/mePWzrSgcxQL2XHKEn+1HvBVrtWevuk/aQCj/JL2qtg"
+                "NJtYpVLHRwGcBJAC5ijq2BUDGbuLoDfqfdBQZl0DPu83pIs49lQsG+29yvl"
+                "flAp8BV7wPOG+HX2Azwh3wqsmthuDeLCM3nVKXNfCw==";
+
+            testValidatorToken(KeyType::ed25519, tokenManifest, tokenBlob);
```

Comment:
I like this test with a valid ed25519 key.  I'm looking for a similar test in this file that uses an secp256k1 key but didn't spot one.  If there isn't one it seems like that would be a good thing to add.

Reply:
There's one with secp256k1 here
https://github.com/ripple/rippled/pull/2647/files/56a9a170c29f9ec593e9f34b81c80d6a86ddd5cc#diff-dd5c6143d81cd734271df4c5fc90e9ebR178